### ISSUE TITLE
web.FileResponse overrides content_type

### ISSUE THREAD

**Issue by thelink2012** [aio-libs/aiohttp#2317]
## Long story short

When returning a `web.FileResponse` from a route, any changes to the `content_type` member is not taken into account when the file is served. Instead, the file extension is always used to determine the content type of the response.

## Expected behaviour

`Content-Type` header of the response be that of `web.FileResponse.content_type` when such has been set.

## Actual behaviour

`Content-Type` header is guessed from the file extension of the served file.

## Steps to reproduce

Create a route that responds something along the lines of

```python
    response = web.FileResponse('/path/to/some/image.png')
    response.content_type = 'application/octet-stream'
    return response
```

then `curl -i` it and notice the `Content-Type` is `image/png`.

## Your environment

```
[~]$ uname -mrs
Linux 4.11.9-1-ARCH x86_64
[~]$ python --version
Python 3.6.2
[~]$ pip show aiohttp | grep Version
Version: 2.2.5
```

**Comment by thelink2012** [comment]
I'm inclined to opening a PR to resolve this issue, suggestions are highly appreciated.

**Comment by asvetlov** [comment]
The only problem is figuring out if content type was explicitly set.
User could do it by `resp.content_type = '...'` and `resp.headers['Content-Type']` modifications.
Default value for property is `'application/octet-stream'`.

Well, the following code should work:
```
if hdrs.CONTENT_TYPE not in self.headers:
        ct, encoding = mimetypes.guess_type(str(filepath))
        if not ct:
            ct = 'application/octet-stream'
        self.content_type = ct
        ....
```

**Comment by thelink2012** [comment]
~I think setting the content type header based on the `path` parameter during `__init__` is a better idea.  This follows the same behaviour as `web.Response`, and gives us access to the guessed content type from the property. As such, this seems like the least suprising way for `web.FileResponse` to behave.~

~Same goes for encoding~.

Ok, I see the flaw behind this, nevermind.

**Comment by asvetlov** [comment]
Fixed by #2319
